extension/src/plots/model/quickPick.ts

```diff
@@ -75,61 +68,41 @@ export const pickCustomPlots = (
   return quickPickManyValues(plotsItems, quickPickOptions)
 }
 
-export const pickCustomPlotType = (): Thenable<CustomPlotType | undefined> => {
-  return quickPickValue(
-    [
-      {
-        description:
-          'A linear plot that compares a chosen metric and param with current experiments.',
-        label: 'Metric Vs Param',
-        value: CustomPlotType.METRIC_VS_PARAM
-      },
-      {
-        description:
-          'A linear plot that shows how a chosen metric changes over selected experiments.',
-        label: 'Checkpoint Trend',
-        value: CustomPlotType.CHECKPOINT
-      }
-    ],
-    {
-      title: Title.SELECT_PLOT_TYPE_CUSTOM_PLOT
-    }
+export const pickCustomPlotType = (
+  columns: Column[],
+  customPlotOrder: CustomPlotsOrderValue[]
+): Thenable<CustomPlotType | undefined> => {
+  const items = []
+  const isMetricVsParamAvailable = checkForMetricVsParamPlotOptions(
+    columns,
+    customPlotOrder
+  )
+  const isCheckpointAvailable = checkForCheckpointPlotOptions(
+    columns,
+    customPlotOrder
   )
-}
-
-const getCustomPlotIds = (
-  customPlotVals: CustomPlotsOrderValue[],
-  customPlotType: CustomPlotType
-): Set<string> => {
-  const plotIds: Set<string> = new Set()
 
-  for (const { type, metric, param } of customPlotVals) {
-    if (type === customPlotType) {
-      plotIds.add(getCustomPlotId(metric, param))
-    }
+  if (isMetricVsParamAvailable) {
+    items.push({
+      description:
+        'A linear plot that compares a chosen metric and param with current experiments.',
+      label: 'Metric Vs Param',
+      value: CustomPlotType.METRIC_VS_PARAM
+    })
   }
 
-  return plotIds
-}
-
-const getCustomPlotPathsFromColumns = (
-  columns: Column[]
-): { metrics: string[]; params: string[] } => {
-  const metrics = []
-  const params = []
-
-  for (const { path, type } of columns) {
-    if (type === ColumnType.METRICS) {
-      metrics.push(
-        removeColumnTypeFromPath(path, ColumnType.METRICS, FILE_SEPARATOR)
-      )
-    } else if (type === ColumnType.PARAMS) {
-      params.push(
-        removeColumnTypeFromPath(path, ColumnType.PARAMS, FILE_SEPARATOR)
-      )
-    }
+  if (isCheckpointAvailable) {
+    items.push({
+      description:
+        'A linear plot that shows how a chosen metric changes over selected experiments.',
+      label: 'Checkpoint Trend',
+      value: CustomPlotType.CHECKPOINT
+    })
   }
-  return { metrics, params }
+
```

Comment:
[Q] Do we need to return early here if there are no items? Even if the path currently cannot be hit it would be as a safeguard.

Reply:
[Q] Are add/remove custom plot available from the command palette? Should they be? Will they be?

Reply:
Same [Q] for experiments table.

Reply:
> but I know we want to avoid adding every possible command in the palette

I don't agree with this. I think we should add as much useful functionality as possible there. Look at what these (very popular) Git extensions do:


https://user-images.githubusercontent.com/37993418/228078301-26e65f7f-92c3-44a1-afee-52addde3870f.mov